tests/functional/tests/smoke.py

```diff
@@ -33,44 +35,51 @@ def install_package(cls) -> None:
         """Install cou package."""
         cou_test_snap = os.environ.get("TEST_SNAP")
         if cou_test_snap:
-            log.info(f"Installing {cou_test_snap}")
-            assert Path(cou_test_snap).is_file()
+            log.info("Installing %s", cou_test_snap)
+            assert Path(cou_test_snap).is_file(), f"{cou_test_snap} is not file"
             # install the snap
-            assert check_call(f"sudo snap install --dangerous {cou_test_snap}".split()) == 0
+            assert (
+                check_call(["sudo", "snap", "install", "--dangerous", cou_test_snap]) == 0
+            ), "cou snap installation failed"
             # connect interfaces
             interfaces_to_connect = [
                 "juju-client-observe",
                 "dot-local-share-cou",
                 "ssh-public-keys",
             ]
             for interface in interfaces_to_connect:
-                assert (
-                    check_call(
-                        f"sudo snap connect charmed-openstack-upgrader:{interface} snapd".split()
-                    )
-                    == 0
+                check_call(
+                    [
+                        "sudo",
+                        "snap",
+                        "connect",
+                        f"charmed-openstack-upgrader:{interface}",
+                        "snapd",
+                    ]
```

Comment:
Why are we removing the assertion for interface connection calls? I think it is useful catching error early; otherwise we would scratch our head wondering why following test cases aren't passing?

Reply:
Actually it was a mistake adding it in the first place. As the [documentation](https://docs.python.org/3/library/subprocess.html#subprocess.check_call) says:
> If the return code was zero then return, otherwise raise [CalledProcessError](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError). The [CalledProcessError](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError) object will have the return code in the [returncode](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError.returncode) attribute.

This means that the assert is never called. Ex:
```
assert check_call(["ls", "asdf"]) == 1, "my custom message"
ls: cannot access 'asdf': No such file or directory
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['ls', 'asdf']' returned non-zero exit status 2.
```
I'll remove all asserts that are using `check_call`

Reply:
Ah, this makes sense. :+1: